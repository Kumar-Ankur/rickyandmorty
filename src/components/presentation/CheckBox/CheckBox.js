import React from "react";
import { getFilteredData } from "../../../utils";

const CheckBox = (props) => {
  const { data, addFilter, removeFilter } = props;

  /**
   * @function handlecheckBox
   * @param { Object } event - window Object for capturing event when user click on checkbox.
   * @description - call getFilteredData function to dispatch the action to update the state based on checkbox selection.
   */
  const handlecheckBox = (event) => {
    getFilteredData(event, addFilter, removeFilter);
  };

  return (
    <>
      <input
        data-test="component-checkbox"
        type="checkbox"
        id={data}
        name={data}
        value={data}
        onClick={(e) => handlecheckBox(e)}
      />
      <label htmlFor={data}>{data}</label>
      <br />
    </>
  );
};

export default CheckBox;
