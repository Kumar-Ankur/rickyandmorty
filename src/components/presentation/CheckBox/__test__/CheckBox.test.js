import React from "react";
import { shallow } from "enzyme";
import CheckBox from "../CheckBox";

/**
 * Factory function to create a shallowwrapper for the App component.
 * @function setUp
 * @param { Object } props - Component props specific to this setup
 * @param { Object } state - Initial state for setUp
 * @returns { shallowWrapper }
 *
 */
function setUp(props = {}) {
  return shallow(<CheckBox {...props} />);
}

/**
 * Return shallowWarpper containing node(s) with the given data-test attribute
 * @function findByTestAttr
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} val - value of data-test attribute for search
 * @returns { shallowWrapper}
 */
function findByTestData(wrapper, val) {
  return wrapper.find(`[data-test='${val}']`);
}

describe("check checkbox component", () => {
  let wrapper;
  const props = {
    data: "Human",
    addFilter: jest.fn(),
    removeFilter: jest.fn()
  };
  beforeEach(() => {
    wrapper = setUp(props);
  });

  test("generate snapshot for checkbox component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("check checkbox component render without crash", () => {
      const componentCheckbox = findByTestData(wrapper, 'component-checkbox');
      expect(componentCheckbox.length).toBe(1);
  });

  test('check onclick event working without error', () => {
      const getFilteredDataMock = jest.fn();
      const event = { target: { checked: true, value: 'test'}}
      const componentCheckbox = findByTestData(wrapper, 'component-checkbox');
      componentCheckbox.simulate('click', event);
      expect(componentCheckbox.props().name).toBe('Human');
  })
});
