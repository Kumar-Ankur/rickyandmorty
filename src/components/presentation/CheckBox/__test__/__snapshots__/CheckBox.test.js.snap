// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check checkbox component generate snapshot for checkbox component 1`] = `ShallowWrapper {}`;
